ddspipe_participants/src/cpp/reader/dds/CommonReader.cpp

```diff
@@ -128,41 +128,42 @@ utils::ReturnCode CommonReader::take_nts_(
         return utils::ReturnCode::RETCODE_NO_DATA;
     }
 
-    RtpsPayloadData* rtps_data;
+    std::unique_ptr<RtpsPayloadData> rtps_data;
```

Comment:
I don't see the purpose of having a unique pointer here. I prefer to always transfer ownership to `data`, so destruction is always done in the same scope.

Reply:
I believe that `data` should only be assigned the value we will actually `take`. To me it feels a bit odd to assign the value of `rtps_data` to `data` when `take_next_sample` returns an error, for example.

Reply:
I guess it's a matter of taste.